fe/fe-core/src/main/java/com/starrocks/external/hive/events/MetastoreEventFactory.java

```diff
@@ -0,0 +1,124 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.external.hive.events;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.starrocks.catalog.Catalog;
+import com.starrocks.catalog.HiveTable;
+import com.starrocks.common.DdlException;
+import com.starrocks.external.hive.HiveMetaCache;
+import com.starrocks.external.hive.HivePartitionKey;
+import org.apache.hadoop.hive.metastore.api.NotificationEvent;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Factory class to create various MetastoreEvents.
+ */
+public class MetastoreEventFactory implements EventFactory {
+    private static final Logger LOG = LogManager.getLogger(MetastoreEventFactory.class);
+
+    /**
+     * For an {@link AddPartitionEvent} and {@link DropPartitionEvent} drop event,
+     * we need to divide it into multiple events according to the number of partitions it processes.
+     * It is convenient for creating batch tasks to parallel processing.
+     */
+    @Override
+    public List<MetastoreEvent> get(NotificationEvent event, HiveMetaCache metaCache, HiveTable table) {
+        Preconditions.checkNotNull(event.getEventType());
+        MetastoreEventType metastoreEventType = MetastoreEventType.from(event.getEventType());
+        switch (metastoreEventType) {
+            case CREATE_TABLE:
+                return CreateTableEvent.getEvents(event, metaCache);
+            case ALTER_TABLE:
+                return AlterTableEvent.getEvents(event, metaCache);
+            case DROP_TABLE:
+                return DropTableEvent.getEvents(event, metaCache);
+            case ADD_PARTITION:
+                return AddPartitionEvent.getEvents(event, metaCache, table.getPartitionColumns());
+            case ALTER_PARTITION:
+                return AlterPartitionEvent.getEvents(event, metaCache);
+            case DROP_PARTITION:
+                return DropPartitionEvent.getEvents(event, metaCache, table.getPartitionColumns());
+            case INSERT:
+                return InsertEvent.getEvents(event, metaCache);
+            default:
+                // ignore all the unknown events by creating a IgnoredEvent
+                return Lists.newArrayList(new IgnoredEvent(event, metaCache));
```

Comment:
Are CreateTableEvent and DropTableEvent unnecessary, and treated as IgnoredEvent?

Reply:
Yes, it should be changed as you suggested. 
In fact, when I wrote this, I planned to support Drop Table, but considering that we are currently creating tables by manually creating external tables. So I plan to support Create Table and Drop table at the same time later. Because we are supporting the synchronization of  hive database  through synchronization tools, I will support Create and Drop Table in the next sprint,  and backward compatibility is reserved here.